src/System.Net.Http/src/System/Net/Http/Unix/CurlTransportContext.cs

```diff
@@ -2,33 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Security.Authentication.ExtendedProtection;
+using System.Threading;
 
 namespace System.Net.Http
 {
-    internal class CurlTransportContext : TransportContext
+    internal sealed class CurlTransportContext : TransportContext
     {
-        private CurlChannelBinding _channelBinding = new CurlChannelBinding();
+        private CurlChannelBinding _channelBinding;
 
         internal CurlChannelBinding CurlChannelBinding
         {
             get
             {
-                return _channelBinding;
+                CurlChannelBinding binding = Volatile.Read(ref _channelBinding);
+                if (binding == null)
+                {
+                    var newBinding = new CurlChannelBinding();
+                    binding = Interlocked.CompareExchange(ref _channelBinding, newBinding, null);
```

Comment:
I see, thanks.  In that case, I've simplified the change.  Since the concurrent usage is only theoretical, I'm using LazyInitializer.EnsureInitialized, which will use almost the exact same pattern I was, except for disposing of the SafeHandle in the case that a thread loses the race (but we don't expect that to occur, and in the extremely rare case it does, it'll be handled by finalization).